src/main.tsx

```diff
@@ -16,24 +13,39 @@ import Privacy from "./routes/Privacy";
 import Terms from "./routes/Terms";
 import Calendar from "./routes/Calendar";
 
-//Event Pages
+// Event Pages
 import OWeek from "./routes/eventPages/OWeek";
 import OWeekGame from "./routes/eventPages/OWeekGame";
 import GameComplete from "./routes/eventPages/GameComplete";
 import HackiethonQuiz from "./routes/eventPages/HackiethonQuiz";
 import QuizResults from "./features/HackiethonQuiz/QuizResults";
 
-//Blog Pages
+// Blog Pages
 import Blogs from "./routes/Blogs";
 import { BlogInterface, BlogsData } from "./routes/blogs/BlogsData";
 
-//Google Analytics
+// Google Analytics
 import ReactGA from "react-ga4";
 import Hackiethon from "./routes/eventPages/Hackiethon";
 import Links from "./routes/Links";
 
 ReactGA.initialize("G-BCJY191MCE");
 
+// Interface for the props that the Redirect component will receive
+interface RedirectProps {
+  to: string;
+}
+
+// Redirect component using the React.FC type
+const Redirect: React.FC<RedirectProps> = ({ to }) => {
+  useEffect(() => {
+    window.location.replace(to);
+  }, [to]);
+
+  return null;
+};
```

Comment:
I think its better to put this in a separate component due to the error currently showing. You can make something like Redirect.tsx inside of the routes folder then import it here instead and that should fix it.